src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPGlobalProperties.cs

```diff
@@ -8,54 +8,24 @@ namespace System.Net.NetworkInformation
 {
     internal class OsxIPGlobalProperties : UnixIPGlobalProperties
     {
-        public override string DhcpScopeName
-        {
-            get
-            {
-                throw new PlatformNotSupportedException();
-            }
-        }
+        public override string DhcpScopeName { get { throw new PlatformNotSupportedException(); } }
 
-        public override string DomainName
-        {
-            get
-            {
-                return HostInformation.DomainName;
-            }
-        }
+        public override string DomainName { get { return HostInformation.DomainName; } }
 
-        public override string HostName
-        {
-            get
-            {
-                return HostInformation.HostName;
-            }
-        }
+        public override string HostName { get { return HostInformation.HostName; } }
 
-        public override bool IsWinsProxy
-        {
-            get
-            {
-                return false;
-            }
-        }
+        public override bool IsWinsProxy { get { throw new PlatformNotSupportedException(); } }
 
-        public override NetBiosNodeType NodeType
-        {
-            get
-            {
-                return NetBiosNodeType.Unknown;
-            }
-        }
+        public override NetBiosNodeType NodeType { get { return NetBiosNodeType.Unknown; } }
 
         public unsafe override TcpConnectionInformation[] GetActiveTcpConnections()
         {
             int realCount = Interop.Sys.GetEstimatedTcpConnectionCount();
-            int infoCount = (int)(realCount * 1.5f);
+            int infoCount = realCount * 2;
             Interop.Sys.NativeTcpConnectionInformation* infos = stackalloc Interop.Sys.NativeTcpConnectionInformation[infoCount];
             while (Interop.Sys.GetActiveTcpConnectionInfos(infos, &infoCount) == -1)
             {
-                var newAlloc = stackalloc Interop.Sys.NativeTcpConnectionInformation[infoCount];
+                Interop.Sys.NativeTcpConnectionInformation* newAlloc = stackalloc Interop.Sys.NativeTcpConnectionInformation[infoCount];
```

Comment:
Is there a potential stack overflow here?  stackalloc isn't going to release the allocated memory on the stack until the end of the function, and we're inside a loop, plus we're allocating for an arbirary size here.